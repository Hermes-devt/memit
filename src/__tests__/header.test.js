

import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../components/header';

//jest cleans up the garbage around the test. 
afterEach(cleanup);

it('Header renders as it should be!', ()=>{
  const {asFragment} = render(<Header />);
  expect( asFragment()).toMatchSnapshot();
});


it( "inserts text in h1", ()=>{
  // jest-dom.
  const {getByTestId, getByText } = render(<Header />)
  expect(getByTestId('h1tag')).toHaveTextContent('temp');
  expect(getByText('temp')).toHaveClass('bla');
})