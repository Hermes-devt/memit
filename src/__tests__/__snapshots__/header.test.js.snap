// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as it should be! 1`] = `
<DocumentFragment>
  <h1
    class="bla"
    data-testid="h1tag"
  >
    temp
  </h1>
</DocumentFragment>
`;
